### DISCUSSION TITLE
(supabase_flutter) user null after responding to email invitation link

### DISCUSSION THREAD

**Discussion by des256** [supabase/supabase#9935] [Questions]
Hi,

I'm inviting a user (with the invite button on app.supabase...). The user receives the appropriate email and clicks on the invitation link. This link directs to my flutter app that does:

```
Supabase supabase = await Supabase.initialize(url: url,anonKey: anonKey);
_supabaseClient = supabase.client;
```

However, the new user is now not logged in (`_supabaseClient.currentUser = null`). The supabase connection is fine, because regular logins in the same app work just fine, and I can access the database.

I'm using supabase_flutter ^1.0.1. I can create a minimal project to recreate this behavior, but maybe I'm overlooking something really simple.

**Comment by dshukertjr** [comment]
@des256 
Where are you calling `_supabaseClient.currentUser`? Currently, the sign in of the user happens synchronously, meaning that even if the user landed on the app with proper login link, there is a split second where the user is null. This has been fixed in [this PR](https://github.com/supabase-community/supabase-flutter/pull/262), but is currently waiting to be released hopefully in the next few days.

**Reply by des256** [reply]
Hi, yes that was the problem. I inserted a blocking wait for the signin auth event, and that seems to solve all the issues for now.
